mp4.go

```diff
@@ -201,7 +640,7 @@ func (m metadataMP4) readAtomData(r io.ReadSeeker, name string, size uint32, pro
 		if len(b) < 1 {
 			return fmt.Errorf("invalid encoding: expected at least %d bytes, for integer tag data, got %d", 1, len(b))
 		}
-		data = getInt(b[:1])
+		data = getInt(b[len(b)-1:])
```

Comment:
Is there some padding in front that we need to jump over?  Might be best to explicitly say that (for future reference!)